app/assets/javascripts/govuk-component/option-select.js

```diff
@@ -0,0 +1,137 @@
+    this.updateClearingLink();
+
+    // Attach event listeners for clearing
+    this.$clearingLink.on('click', this.resetOptions.bind(this));
+    this.$options.on('click', this.updateClearingLink.bind(this));
+
+    // Performance in ie 6/7 is not good enough to support animating the opening/closing
+    // so do not allow option-selects to be collapsible in this case
+    var allowCollapsible = (typeof ieVersion == "undefined" || ieVersion > 7) ? true : false;
+    if(allowCollapsible){
+
+      // Add js-collapsible class to parent for CSS
+      this.$optionSelect.addClass('js-collapsible');
+
+      // Add the little arrow toggle image
+      this.attachOpenCloseToggle();
+
+      // Add open/close listeners
+      this.$optionSelect.find('.container-head').on('click', this.toggleOptionSelect.bind(this));
+
+      // Add key listeners the option-select is fully usable with a keyboard
+      this.$optionSelect.on('focus', this.listenForKeys.bind(this));
+      this.$optionSelect.on('blur', this.stopListeningForKeys.bind(this));
+
+      // Add a listener to the checkboxes so if you navigate to them with the keyboard you can definitely see them
+      this.$options.on('focus', this.ensureOptionSelectIsOpen.bind(this));
+
+      this.close();
+    }
+  }
+
+  OptionSelect.prototype.attachClearingLink = function attachClearingLink(){
+    this.$optionSelect.find('.container-head').append('<a class="js-clear-selected">clear</a>');
+    return this.$optionSelect.find('.js-clear-selected');
+  };
+
+  OptionSelect.prototype.resetOptions = function resetOptions(){
+    this.$options.prop({
+      indeterminate: false,
+      "checked": false
+    }).trigger("change");
+    this.$clearingLink.addClass('js-hidden');
+    return false;
+  };
+
+  OptionSelect.prototype.updateClearingLink = function updateClearingLink(){
+    var anyOptions = this.$options.is(":checked"),
+        clearingLinkHidden = this.$clearingLink.hasClass('js-hidden');
+
+    if (anyOptions && clearingLinkHidden) {
+      this.$clearingLink.removeClass('js-hidden');
+    } else if (!anyOptions && !clearingLinkHidden) {
+      this.$clearingLink.addClass('js-hidden');
+    }
+  };
+
+  OptionSelect.prototype.attachOpenCloseToggle = function attachOpenCloseToggle(){
+    this.$optionSelect.find('.container-head').append('<div class="js-toggle"></div>');
+  };
+
+  OptionSelect.prototype.toggleOptionSelect = function toggleOptionSelect(){
+    if (this.isClosed()) {
+      this.open();
+    } else {
+      this.close();
+    }
+  };
+
+  OptionSelect.prototype.open = function open(){
+    this.$optionSelect.removeClass('js-closed');
+    this.setupHeight();
+  };
+
+  OptionSelect.prototype.close = function close(){
+    this.$optionSelect.addClass('js-closed');
+  };
+
+  OptionSelect.prototype.isClosed = function isClosed(){
+    return this.$optionSelect.hasClass('js-closed');
+  };
+
+  OptionSelect.prototype.ensureOptionSelectIsOpen = function ensureOptionSelectIsOpen(){
```

Comment:
Is it worth adding the `if (this.isClosed()) {` to the contents of `open` and removing this method? It would make for a simpler API and anything wanting to ensure the OptionSelect is open can just call open and not have to worry about any side effects.